lib/Serialization/Deserialization.cpp

```diff
@@ -4324,6 +4324,11 @@ Type ModuleFile::getType(TypeID TID) {
     return nullptr;
   }
 
+#ifndef NDEBUG
+  PrettyStackTraceType(ctx, "deserializing", typeOrOffset.get());
+  assert(!typeOrOffset.get()->hasError());
```

Comment:
I suspect we're getting an ErrorType because a mapTypeIntoContext() is failing because a generic environment was not (de)serialized properly. cc @DougGregor